test/unit/webgl/p5.Texture.js

```diff
@@ -12,15 +15,27 @@ suite('p5.Texture', function() {
   setup(function(done) {
     myp5 = new p5(function(p) {
       p.preload = function() {
+        // texImg2 must have powerOfTwo dimensions
         texImg2 = p.loadImage('unit/assets/target.gif');
-
+        // texImg3 must NOT have powerOfTwo dimensions
+        texImg3 = p.loadImage('unit/assets/nyan_cat.gif');
         // texture object isn't created until it's used for something:
         //p.box(70, 70, 70);
       };
       p.setup = function() {
         canvas = p.createCanvas(100, 100, p.WEBGL);
         texImg1 = p.createGraphics(2, 2, p.WEBGL);
         p.texture(texImg1);
+        p.createImg(texImg2.canvas.toDataURL(), '', 'anonymous', el => {
+          el.size(50, 50);
+          imgElementPowerOfTwo = el;
+          p.texture(imgElementPowerOfTwo);
```

Comment:
It looks like this is working right now, but maybe we can make it a tad clearer for readers of the tests which texture will be active at the end of `setup`? I think the success callback here will always run immediately due to it being a data URL, but I think `done()` gets called before those callbacks run. Maybe we could do something like this to ensure we always end with `texImg1` as the active texture, and that we call `done` after?

```js
new Promise(resolve => {
  p.createImg(texImg2.canvas.toDataURL(), '', 'anonymous', el => {
    el.size(50, 50);
    imgElementPowerOfTwo = el;
    p.texture(imgElementPowerOfTwo);
    resolve();
  });
}).then(() => new Promise(resolve => {
  p.createImg(texImg3.canvas.toDataURL(), '', 'anonymous', el => {
    el.size(50, 50);
    imgElementNotPowerOfTwo = el;
    p.texture(imgElementNotPowerOfTwo);
    resolve();
  });
})).then(() => {
  p.texture(texImg1);
  done();
});
```

Reply:
sure thing--added this.